ext/lb/haproxy/generate.go

```diff
@@ -0,0 +1,170 @@
+package haproxy
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/ehazlett/interlock/ext/lb/utils"
+	"github.com/samalba/dockerclient"
+)
+
+func (p *HAProxyLoadBalancer) GenerateProxyConfig(containers []dockerclient.Container) (interface{}, error) {
+	var hosts []*Host
+
+	proxyUpstreams := map[string][]*Upstream{}
+	hostChecks := map[string]string{}
+	hostBalanceAlgorithms := map[string]string{}
+	hostBackendOptions := map[string][]string{}
+	hostSSLOnly := map[string]bool{}
+	hostSSLBackend := map[string]bool{}
+	hostSSLBackendTLSVerify := map[string]string{}
+
+	networks := map[string]string{}
+
+	// TODO: instead of setting defaults here use
+	// SetDefaultConfig in the utils package
+	for _, cnt := range containers {
+		cntId := cnt.Id[:12]
+		// load interlock data
+		cInfo, err := p.client.InspectContainer(cntId)
+		if err != nil {
+			return nil, err
+		}
+
+		hostname := utils.Hostname(cInfo.Config)
+		domain := utils.Domain(cInfo.Config)
+
+		if domain == "" {
+			continue
+		}
+
+		if hostname != domain && hostname != "" {
+			domain = fmt.Sprintf("%s.%s", hostname, domain)
+		}
+
+		healthCheck := utils.HealthCheck(cInfo.Config)
+		healthCheckInterval, err := utils.HealthCheckInterval(cInfo.Config)
+		if err != nil {
+			log().Errorf("error parsing health check interval: %s", err)
+			continue
+		}
+
+		if healthCheck != "" {
+			if val, ok := hostChecks[domain]; ok {
+				// check existing host check for different values
+				if val != healthCheck {
+					log().Warnf("conflicting check specified for %s", domain)
+				}
+			} else {
+				hostChecks[domain] = healthCheck
+				log().Debugf("using custom check for %s: %s", domain, healthCheck)
+			}
+
+			log().Debugf("check interval for %s: %d", domain, healthCheckInterval)
+		}
+
+		hostBalanceAlgorithms[domain] = utils.BalanceAlgorithm(cInfo.Config)
+
+		backendOptions := utils.BackendOptions(cInfo.Config)
+
+		if len(backendOptions) > 0 {
+			hostBackendOptions[domain] = backendOptions
+			log().Debugf("using backend options for %s: %s", domain, strings.Join(backendOptions, ","))
+		}
+
+		hostSSLOnly[domain] = utils.SSLOnly(cInfo.Config)
+
+		// ssl backend
+		hostSSLBackend[domain] = utils.SSLBackend(cInfo.Config)
+		hostSSLBackendTLSVerify[domain] = utils.SSLBackendTLSVerify(cInfo.Config)
+
+		addr := ""
+
+		// check for networking
+		if n, ok := utils.OverlayEnabled(cInfo.Config); ok {
+			log().Debugf("configuring docker network: name=%s", n)
+
+			// FIXME: for some reason the request from dockerclient
+			// is not returning a populated Networks object
+			// so we hack this by inspecting the Network
+			// we should switch to engine-api/client -- hopefully
+			// that will fix
+			//net, found := cInfo.NetworkSettings.Networks[n]
```

Comment:
If there was a populated Networks object, would you even need the `interlock.network` label on a container?

You could just a populated Networks object to discover the network(s) of a container and the associated address(es).

It seems redundant to have to specify the network twice when running containers like

```
docker run --detach \
  --expose 8080 \
  --net mynetwork \
  --label interlock.network=mynetwork \
  --label interlock.hostname=test \
  --label interlock.domain=local \
  ehazlett/docker-demo
```

I suppose it's a moot point samalba/dockerclient gets replaced.